src/flow/src/transform/literal.rs

```diff
@@ -105,7 +175,107 @@ pub(crate) fn from_substrait_literal(lit: &Literal) -> Result<(Value, CDT), Erro
             )
         }
         Some(LiteralType::Null(ntype)) => (Value::Null, from_substrait_type(ntype)?),
-        _ => not_impl_err!("unsupported literal_type")?,
+        Some(LiteralType::IntervalDayToSecond(interval)) => {
+            let (days, seconds, microseconds) =
+                (interval.days, interval.seconds, interval.microseconds);
+            let millis = microseconds / 1000 + seconds * 1000;
+            let value_interval = common_time::Interval::from_day_time(days, millis);
+            (
+                Value::Interval(value_interval),
+                CDT::interval_day_time_datatype(),
+            )
+        }
+        Some(LiteralType::IntervalYearToMonth(interval)) => (
+            Value::Interval(common_time::Interval::from_year_month(
+                interval.years * 12 + interval.months,
+            )),
+            CDT::interval_year_month_datatype(),
+        ),
+        Some(LiteralType::UserDefined(UserDefined {
+            val: Some(Val::Value(val)),
+            type_reference,
+            ..
+        })) => {
+            fn from_bytes<T: FromBytes>(i: &Bytes) -> Result<T, Error>
+            where
+                for<'a> &'a <T as num_traits::FromBytes>::Bytes:
+                    std::convert::TryFrom<&'a [u8], Error = TryFromSliceError>,
+            {
+                let (int_bytes, _rest) = i.split_at(std::mem::size_of::<T>());
+                let i = T::from_le_bytes(int_bytes.try_into().map_err(|e| {
+                    UnexpectedSnafu {
+                        reason: format!(
+                            "Expect slice to be {} bytes, found {} bytes, error={:?}",
+                            std::mem::size_of::<T>(),
+                            int_bytes.len(),
+                            e
+                        ),
+                    }
+                    .build()
+                })?);
+                Ok(i)
+            }
+            // see https://github.com/apache/datafusion/blob/146b679aa19c7749cc73d0c27440419d6498142b/datafusion/substrait/src/logical_plan/producer.rs#L1957
+            // for interval type's transform to substrait
+            match *type_reference {
```

Comment:
Such a large method. It would be better to add a method to handle the interval type.